### ISSUE TITLE
[env][documentation] cliff_walking.rs: DiscreteAction::from_index panics, undocumented

### ISSUE THREAD

**Issue by anthonytorlucci** [anthonytorlucci/rlevo#1093]
**File:** `crates/rlevo-environments/src/toy_text/cliff_walking.rs:235-251` (review §5.2)

```rust
impl DiscreteAction<1> for CliffWalkingAction {
    const ACTION_COUNT: usize = 4;

    fn from_index(index: usize) -> Self {
        match index {
            0 => CliffWalkingAction::Up,
            1 => CliffWalkingAction::Right,
            2 => CliffWalkingAction::Down,
            3 => CliffWalkingAction::Left,
            _ => panic!("CliffWalkingAction index {index} out of range [0, 4)"),
        }
    }
```

Panic-on-bad-index is the documented `DiscreteAction::from_index` contract
(`crates/rlevo-core/src/action.rs:117-123`), so this is not a bug today. But the
impl carries **no doc comment at all** — no `# Panics` section — and there is no
fallible sibling for callers whose index comes from data (a replay log, a
deserialized trajectory, a mis-sized policy head) rather than from a trusted
`argmax` over `ACTION_COUNT` logits.

**Why this is filed now.** This is the fourth instance of a shape already filed
three times — #954 (blackjack), #951 (taxi), #971 (frozen_lake) — and the only
one without its own issue. The toy_text review's cliff_walking summary row
(`.scratchpad/code-review-30-06-2026/code-review-01-07-2026-env-toy-text.md:606`)
currently points at #951, which is titled for `taxi.rs`; that row needs a
truthful pointer.

**Suggested change:** replicate the precedent landed for `GridAction` in
`d2b423b` (issue #813) — an inherent
`try_from_index(usize) -> Result<Self, InvalidActionError>`, an
`impl TryFrom<usize>` delegating to it, and `from_index` delegating via
`unwrap_or_else(|_| panic!(...))` with a `# Panics` doc section. The existing
panic string is preserved verbatim; there are no `#[should_panic(expected = ...)]`
tests anywhere under `crates/rlevo-environments/src/toy_text/` constraining it.

**Risk:** Low — purely additive, no behavior change for existing callers.

Related: #954, #951, #971 (same shape, same family), #813 (the landed precedent).
